pkg/sql/pgwire/pgerror/errors.go

```diff
@@ -132,6 +133,39 @@ func GetPGCause(err error) (*Error, bool) {
 	}
 }
 
+// NewAssertionErrorf creates an internal error.
+func NewAssertionErrorf(format string, args ...interface{}) error {
+	err := NewErrorWithDepthf(1, CodeInternalError, "programming error: "+format, args...)
+	err.InternalCommand = captureTrace()
+	err.Detail = err.InternalCommand
+	err.Hint = "You have encountered an error inside CockroachDB.\n\n" +
+		"Don't worry! Your data is likely safe.\n" +
+		"Please report this error with details at:\n\t https://github.com/cockroachdb/cockroach/issues/new/choose\nor\n\t support@cockroachlabs.com.\n\n" +
+		"Thank you!"
+	return err
```

Comment:
we have a ton more info right now (the full stack trace) than we will in a string->int map at reporting time. A sentry event has much, much more useful information for debugging. I'd almost like to send a sentry event right here, the way we would if this were a crash, since that has the best chance of being actionable for fixing the issue.

*BUT* I _don't_ want to send a sentry every 1000 times per second if this ends up in a hot path. Maybe we could have a "when first hit" event?